app/controllers/corporate_controller.rb

```diff
@@ -18,7 +18,8 @@ def show
   end
 
   def create
-    @partner = CorporatePartner.new(partner_params)
+    @partner = CorporatePartner.find_or_initialize_by(params[:email])
```

Comment:
That is odd.

I noticed in your commit that you changed `params[:email]` to `params[:corporate_partner][:email]` — which would suggest that `params[:email]` might have returned `nil`.

`Model.find_or_initialize_by(nil)` will return the first record in `Model` — so good we caught that! The reason I go into detail is — maybe this is a sign the tests here need tweaking to make thm more observant?